### PR TITLE
The user is set on the login screen if transmitted from the client

### LINKED ISSUE #66: The login page do not suggest the user despite the `user=` in the URL
Going to
`index.php/apps/oauth2/authorize?response_type=code&client_id=xdXOt13JKxym1B1QcEncf2XDkLAexMBFwiT9j6EfhhHFJhs2KM9jbjTmf8JBXE69&redirect_uri=http://localhost:44783&user=alice`

Redirects to 

`index.php/login?redirect_url=%252F%257Eowncloud%252Findex.php%252Fapps%252Foauth2%252Fauthorize%253Fresponse_type%253Dcode%2526client_id%253DxdXOt13JKxym1B1QcEncf2XDkLAexMBFwiT9j6EfhhHFJhs2KM9jbjTmf8JBXE69%2526redirect_uri%253Dhttp%253A%252F%252Flocalhost%253A44783%2526user%253Dalice`

Which does not pre-fill the username  (and ideally makes it read-only)

I suppose the same trick as the one used in 9400b663c9854c073a1ca06f1ea01428a496350b should be done to see the `user=` parametter in the redirect_url

### PR DESCRIPTION
**Pull request by DeepDiver1975**
fixes #66

### CODE REVIEW

File: js/login.js [hunk 1]
```diff
@@ -21,8 +21,13 @@
...
 	var $loginMessage = $('#body-login').find('#message');
 	if ($loginMessage.length) {
-		var client = $("data[key='oauth2']").attr('value');
+		var client = $("data[key='oauth2']").attr('client');
 		var msg = t('oauth2', 'The application "{app}" is requesting access to your account. To authorize it, please log in first.', {app : client});
 		$loginMessage.parent().append('<div class="warning"><div class="icon-info-white" />'+msg+'</div>');
+		var user = $("data[key='oauth2']").attr('user');
+		if (user) {
+			$('#password').val('');
+			$('#user').val(user);
```
Comment (SamuAlfageme) [inline js/login.js]: @DeepDiver1975 what do you think of including:

```
$('#user').prop('readonly', true);
```

... so we force the user to login as the original user?

We'll loose focus on the login form though; password should receive it now for better UX I'm guessing.

Comment (DeepDiver1975) [inline js/login.js]: > I'm still getting a redirection and a code with this PR when logged in as a different user and doing the switch via <server>/index.php/apps/oauth2/logout?user=<user>&requesttoken=<token>&response_type=code&client_id=<client_id>&redirect_uri=<redirect_url> (same as described in #66 (comment)) I guess it's in the PR's scope, no?

I still need to work on this ... maybe in scope .... lets see ...

**Comment by DeepDiver1975** [comment]
> $('#user').prop('readonly', true);
> 
> ... so we force the user to login as the original user?

both taken care of ....

**Comment by SamuAlfageme** [comment]
Very nice! Working flawlessly together with https://github.com/owncloud/core/pull/28511 👍 

@DeepDiver1975 move https://github.com/owncloud/oauth2/pull/67#pullrequestreview-53183789 to a different issue and merge here?

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975 move #67 (review) to a different issue and merge here?

yes please - I need to understand this better ... THX

**Comment by SamuAlfageme** [comment]
> yes please - I need to understand this better ... THX

Done in https://github.com/owncloud/oauth2/issues/68. Hope it helps!

**Pull request merged**